File: 4차 과제/5-2/py 소스코드/5-2.py
from header5_2 import Stack
from func5_2_1 import LinkdedStack
from func5_2_2 import print_func

def main():
    list=LinkdedStack()
    print("** 연결 스택 연산 **")
    print_func(list)

    list.push_func(1)
    print_func(list)

    list.push_func(2)
    print_func(list)

    list.push_func(3)
    print_func(list)

    print_func(list, False)
    print(f" peek=>{list.top.stack}")

    pop_value=list.pop_func()
    print_func(list, False)
    print(f"   pop=>{pop_value}")

    pop_value=list.pop_func()
    print_func(list, False)
    print(f"     pop=>{pop_value}")

    pop_value=list.pop_func()
    print_func(list, False)
    print(f"      pop=>{pop_value}")

if __name__ == "__main__":
    main()

File: 4차 과제/5-2/C 소스코드/5-2.c
#include <stdio.h>
#include "5-2.h"

int main() {
    elment item;
    top=NULL;
    printf("**연결 스택 연산**\n");
    
    print_func();
    printf("\n");

    push(1);
    print_func();
    printf("\n");

    push(2);
    print_func();
    printf("\n");

    push(3);
    print_func();
    printf("\n");

    item=peek();
    print_func();
    printf("\t peek => %d\n", item);


    item=pop();
    print_func();
    printf("\t pop => %d\n", item);

    item=pop();
    print_func();
    printf("\t pop => %d\n", item);

    item=pop();
    print_func();
    printf("\t\t pop => %d\n", item);
    return 0;
}